src/schemas/Program/index.js

```diff
@@ -388,66 +363,78 @@ const resolvers = {
 				genomicDonors: 0,
 			};
 
-			try {
-				const createResponse = await programService.createProgram(program, egoToken);
-				return resolvePrivateSingleProgram(egoToken, program.shortName);
-			} catch (err) {
-				const GRPC_INVALID_ARGUMENT_ERROR_CODE = 3;
-				if (err.code === GRPC_INVALID_ARGUMENT_ERROR_CODE) {
-					//this just wraps it into standard apollo semantics
-					throw new UserInputError(err);
-				} else {
-					throw new ApolloError(err);
-				}
-			}
+			const createResponse = await programService.createProgram(program, egoToken);
+			return resolvePrivateSingleProgram(egoToken, program.shortName);
 		},
 
 		updateProgram: async (obj, args, context, info) => {
+			// extract information from query parameters
 			const { egoToken } = context;
 			const updates = pickBy(get(args, 'updates', {}), (v) => v !== undefined);
-			const shortName = get(args, 'shortName', {});
-
+			const programShortName = get(args, 'shortName', {});
+			const dataCenterShortName = args.updates.dataCenterShortName;
+			delete updates.dataCenterShortName;
+
+			//make a request to get dataCenter data using dataCenterShortName (from query paramenter, updates) and format response
+			const [dataCenterResponse] = await programService.listDataCenters(
```

Comment:
is the return value of `listDataCenters` an iterable?

Reply:
Yes it is returning an array or throwing an error

Reply:
Sorry I'm trying to pin down what I'm confused on here. 

it's returning an array but I'm confused why it's returning an array.
It's just destructing the first element, and throwing away any other elements.
Then destructing again on line #383 to just use the same object in the `combinedUpdates` 

Not saying this is a blocker. Just wondering if there's a reason we're getting more data than we need?